test/files/run/wacky-value-classes.scala

```diff
@@ -0,0 +1,19 @@
+// scala/bug#9240
+final class AnyValNothing(val self: Nothing) extends AnyVal
+final class AnyValNull   (val self: Null   ) extends AnyVal
+// scala/bug#10361
+final class AnyValUnit   (val self: Unit   ) extends AnyVal
+
+object Test extends App {
+  def avn = new AnyValNull(null)
+  assert(avn == avn)
+  /*this throws NPE right now b/c null.hashCode() does...*/
+  //assert(avn.hashCode() == 0)
```

Comment:
Switch this (and avu below) to `.##`. If this here returns 0 rather than throw an NPE then you also fixed https://github.com/scala/bug/issues/7396 with this patch!